sound/soc/codecs/rt1308-sdw.c

```diff
@@ -186,6 +186,7 @@ static int rt1308_io_init(struct device *dev, struct sdw_slave *slave)
 		goto _io_init_err_;
 
 	if (rt1308->first_init) {
+		regcache_mark_dirty(rt1308->regmap);
```

Comment:
sorry, I don't understand the connection with the resume process described in the commit message.

first_init is only set once (duh!) during the initial probe?

What am I missing?

Reply:
The resume function will do regcache_sync function.
However, the unattached->attached state could trigger to re-do the io_init function during resume process.
I want to force regcache_sync if the io_init function re-do again.

Reply:
ok, got it now. Somehow 'first_init' made me thing this is something we do only once, but that's not correct, it's done after each call. Maybe we need to change the name of this variable

Also shouldn't we mark this cache dirty at a later point. I am not sure if it make sense to mark the cache as dirty and immediately do a bypass? You'd want to mark it as dirty and do a sync after all the calibration and direct access to the hardware is done?

And last do we need this for rt715 and rt5682?

Reply:
> You'd want to mark it as dirty and do a sync after all the calibration and direct access to the hardware is done?
Yes, but now I want to change the position of regcache_mark_dirty API to suspend function. It be called in the suspend function that this position should be the normal approach in other codec driver.
I will make the patches for 700/711/715/1308. The rt5682 was already did.